snippets/swatch.liquid

```diff
@@ -0,0 +1,31 @@
+{% comment %}
+  Renders a swatch component.
+  Accepts:
+  - swatch: {Object} a swatch object
+  - shape: {String} swatch shape. Accepts 'square', defaults to circle.
+
+  Usage:
+  {% render 'swatch',
+    swatch: value.swatch
+    shape: 'square'
+  %}
+{% endcomment %}
+
+{%- liquid
+  assign swatch_value = nil
+  if swatch.image
+    assign image_url = swatch.image | image_url: width: 50
+    assign swatch_value = 'url(' | append: image_url | append: ')'
+  elsif swatch.color
+    assign swatch_value = 'rgb(' | append: swatch.color.rgb | append: ')'
+  endif
+-%}
```

Comment:
And this we needed here so it can be easily reusable right ? Cause we also have the similar declaration in `product-variant-options.liquid`

Reply:
The declaration in `product-variant-options.liquid` is needed specifically for the dropdown to work, because we need to dynamically update the swatch value on every dropdown change.

The block in `swatch.liquid` is reusable, but it's static